File: ProcessEngine/core/src/main/java/nl/adaptivity/process/engine/HProcessInstance.java
/*
 * Copyright (c) 2016.
 *
 * This file is part of ProcessManager.
 *
 * ProcessManager is free software: you can redistribute it and/or modify it under the terms of version 3 of the
 * GNU Lesser General Public License as published by the Free Software Foundation.
 *
 * ProcessManager is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License along with Foobar.  If not,
 * see <http://www.gnu.org/licenses/>.
 */

package nl.adaptivity.process.engine;

import nl.adaptivity.process.ProcessConsts.Engine;
import nl.adaptivity.xml.XmlDeserializer;
import nl.adaptivity.xml.XmlDeserializerFactory;
import nl.adaptivity.util.xml.XmlUtil;
import nl.adaptivity.xml.XmlException;
import nl.adaptivity.xml.XmlReader;

import javax.xml.namespace.QName;

@XmlDeserializer(HProcessInstance.Factory.class)
public final class HProcessInstance extends XmlHandle {

  public static class Factory implements XmlDeserializerFactory<HProcessInstance> {

    @Override
    public HProcessInstance deserialize(final XmlReader in) throws XmlException {
      return HProcessInstance.deserialize(in);
    }
  }

  public static final java.lang.String ELEMENTLOCALNAME = "instanceHandle";
  public static final QName ELEMENTNAME = new QName(Engine.NAMESPACE, ELEMENTLOCALNAME, Engine.NSPREFIX);

  public HProcessInstance() {
    super(-1);
  }

  public HProcessInstance(final long handle) {
    super(handle);
  }

  private static HProcessInstance deserialize(final XmlReader in) throws XmlException {
    return XmlUtil.deserializeHelper(new HProcessInstance(), in);
  }

  @Override
  public QName getElementName() {
    return ELEMENTNAME;
  }

  @Override
  public boolean equals(final Object obj) {
    return (obj == this) || ((obj instanceof HProcessInstance) && (getHandle() == ((HProcessInstance) obj).getHandle()));
  }

  @Override
  public int hashCode() {
    return (int) getHandle();
  }

}

File: ProcessEngine/core/src/main/kotlin/nl/adaptivity/process/engine/HProcessInstance.kt
/*
 * Copyright (c) 2016.
 *
 * This file is part of ProcessManager.
 *
 * ProcessManager is free software: you can redistribute it and/or modify it under the terms of version 3 of the
 * GNU Lesser General Public License as published by the Free Software Foundation.
 *
 * ProcessManager is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License along with ProcessManager.  If not,
 * see <http://www.gnu.org/licenses/>.
 */

package nl.adaptivity.process.engine

import net.devrieze.util.ComparableHandle
import net.devrieze.util.getInvalidHandle
import net.devrieze.util.security.SecureObject
import nl.adaptivity.process.ProcessConsts.Engine
import nl.adaptivity.xmlutil.*
import javax.xml.namespace.QName

@XmlDeserializer(HProcessInstance.Factory::class)
class HProcessInstance(handle: ComparableHandle<SecureObject<ProcessInstance>>) : XmlHandle<SecureObject<ProcessInstance>>(handle) {

  class Factory : XmlDeserializerFactory<HProcessInstance> {

    @Throws(XmlException::class)
    override fun deserialize(reader: XmlReader): HProcessInstance {
      return HProcessInstance.deserialize(reader)
    }
  }

  constructor() : this(getInvalidHandle<SecureObject<ProcessInstance>>())

  override val elementName: QName
    get() = ELEMENTNAME

  override fun equals(other: Any?): Boolean {
    return other === this || other is HProcessInstance && handleValue == other.handleValue
  }

  override fun hashCode(): Int {
    return handleValue.toInt()
  }

  companion object {

    const val ELEMENTLOCALNAME = "instanceHandle"
    
    @JvmStatic
    val ELEMENTNAME = QName(Engine.NAMESPACE, ELEMENTLOCALNAME, Engine.NSPREFIX)

    @Throws(XmlException::class)
    private fun deserialize(xmlReader: XmlReader): HProcessInstance {
      // For some reason a transactiontype is needed here even though it is dropped
      return HProcessInstance().deserializeHelper(xmlReader)
    }
  }

}
